anndata/_core/anndata.py

```diff
@@ -594,14 +594,20 @@ def _init_as_actual(
 
     def __sizeof__(self, show_stratified=None) -> int:
         def get_size(X):
-            if issparse(X):
-                X_csr = csr_matrix(X)
+            def csr_to_bytes(X_csr):
                 return X_csr.data.nbytes + X_csr.indptr.nbytes + X_csr.indices.nbytes
+
+            if isinstance(X, h5py._hl.dataset.Dataset):
+                return np.array(X.shape).prod() * X.dtype.itemsize
+            elif isinstance(X, sparse._csr.csr_matrix):
+                return csr_to_bytes(X)
+            elif isinstance(X, BaseCompressedSparseDataset):
+                return csr_to_bytes(X._to_backed())
             else:
                 return X.__sizeof__()
 
         size = 0
-        attrs = list(["_X", "_obs", "_var"])
+        attrs = list(["X", "_obs", "_var"])
```

Comment:
Hm, are you sure the `_` wasn’t there for a reason? Is there a test that checks if this works in case `X` is None and so on?